src/rust/engine/process_execution/src/remote.rs

```diff
@@ -1367,7 +1372,13 @@ pub async fn check_action_cache(
             ..remexec::GetActionResultRequest::default()
           };
           let request = apply_headers(Request::new(request), &context.build_id);
-          async move { client.get_action_result(request).await }
+          async move {
+            let lookup_fut = client.get_action_result(request);
+            let timeout_fut = tokio::time::timeout(timeout_duration, lookup_fut);
+            timeout_fut
+              .await
+              .unwrap_or_else(|_| Err(Status::unavailable("Pants client timeout")))
+          }
```

Comment:
This will time out the action request, but not the eager fetch (i.e., downloading the entry contents)... which I assume is the bulk of the traffic to the server: https://github.com/pantsbuild/pants/blob/949c307917c6131aec3cc0442ae1f36e70c14497/src/rust/engine/process_execution/src/remote.rs#L1399-L1425